erpnext/regional/india/e_invoice/utils.py

```diff
@@ -171,10 +171,15 @@ def get_item_list(invoice):
 		item.description = sanitize_for_json(d.item_name)
 
 		item.qty = abs(item.qty)
-		item.discount_amount = 0
-		item.unit_rate = abs(item.base_net_amount / item.qty)
-		item.gross_amount = abs(item.base_net_amount)
-		item.taxable_value = abs(item.base_net_amount)
+
+		if invoice.apply_discount_on == 'Net Total' and invoice.discount_amount:
+			item.discount_amount = abs(item.base_amount - item.base_net_amount)
+		else:
+			item.discount_amount = 0
+
+		item.unit_rate = abs((item.taxable_value - item.discount_amount)/ item.qty)
+		item.gross_amount = abs(item.taxable_value + item.discount_amount)
```

Comment:
```suggestion
		item.gross_amount = abs(item.taxable_value) + item.discount_amount
```

Consider `item.taxable_value = -200` & calculated `item.discount_amount = 20` then `gross_amount = 180` which is wrong.

Tested locally, this creates a mismatch.